access-checkout/src/main/java/com/worldpay/access/checkout/validation/listeners/text/PanTextWatcher.kt

```diff
@@ -209,6 +210,8 @@ internal class PanTextWatcher(
      * rule of the new card brand
      */
     private fun handleCardBrandChange(newCardBrand: RemoteCardBrand?) {
+        if (isPanRequiredLength()) logCardBrands(newCardBrand)
```

Comment:
The future call to the card bin API and handling the card brand change should be decorrelated because they won't happen at the same time, from a user interaction perspective
- card brand changes happen when the user enters the first few digits
- dealing with the card bin api will happen when 12 digits have been entered or when going from 12+ digits to less than 12

So point is that logic should be put in a separate function.

We're incrementally building the base for the future work that will talk to the Card bin API that's why we want to build it right and keeping in mind the work we have left to do (just trying to explain you my way of thinking so you see where this comment is coming from)